### PR TITLE
AP_Param: align info structs to save flash

### PR DESCRIPTION
**Pull request by IamPete1**
This saves a bunch of flash due to better packing of the struct.

### CODE REVIEW

**Comment by andyp1per** [comment]
@IamPete1 you will need to carefully test this with the OSD menu stuff. Also does this change the indexing at all? The OSD menu stuff has a persistent representation that relies on the indexing - so people's config might get broken by this. It may all be fine - just saying it needs careful consideration and testing

**Comment by IamPete1** [comment]
OSD stuff is un-affected. Tested configuring some params none default param on master and then checked there still there on this branch and can still be configured as before.

**Comment by IamPete1** [comment]
Durandal:
```
Binary Name      Text [B]          Data [B]     BSS (B)        Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  ----------------  -----------  -------------  ----------------------------  -------------------------
ardusub          -4872 (-0.3065%)  0 (0.0000%)  0 (0.0000%)    -4872 (-0.3060%)                                 379036
ardurover        -5648 (-0.3433%)  0 (0.0000%)  0 (0.0000%)    -5648 (-0.3428%)                                 324352
blimp            -3804 (-0.2914%)  0 (0.0000%)  -4 (-0.0015%)  -3804 (-0.2910%)                                 662436
arducopter-heli  -6492 (-0.3600%)  0 (0.0000%)  -4 (-0.0015%)  -6492 (-0.3593%)                                 165840
antennatracker   -3616 (-0.2645%)  0 (0.0000%)  0 (0.0000%)    -3616 (-0.2642%)                                 600864
arducopter       -6240 (-0.3473%)  0 (0.0000%)  0 (0.0000%)    -6240 (-0.3466%)                                 172200
arduplane        -6868 (-0.3846%)  0 (0.0000%)  -4 (-0.0015%)  -6868 (-0.3839%)                                 183768
```

MatekF405:
```
Binary Name      Text [B]          Data [B]     BSS (B)        Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  ----------------  -----------  -------------  ----------------------------  -------------------------
ardusub          -3532 (-0.4439%)  0 (0.0000%)  4 (+0.0031%)   -3532 (-0.4433%)                                 189724
ardurover        -4140 (-0.4971%)  0 (0.0000%)  -4 (-0.0031%)  -4140 (-0.4965%)                                 153304
blimp            -2648 (-0.4062%)  0 (0.0000%)  0 (0.0000%)    -2648 (-0.4057%)                                 332980
arducopter-heli  -4668 (-0.5041%)  0 (0.0000%)  4 (+0.0031%)   -4668 (-0.5028%)                                  59248
antennatracker   -2492 (-0.3942%)  0 (0.0000%)  4 (+0.0031%)   -2492 (-0.3937%)                                 352552
arducopter       -4412 (-0.4813%)  0 (0.0000%)  -4 (-0.0031%)  -4412 (-0.4800%)                                  68360
arduplane        -5256 (-0.5498%)  0 (0.0000%)  0 (0.0000%)    -5256 (-0.5485%)                                  30032
```

Pixhawk1-1M:
```
Binary Name      Text [B]          Data [B]     BSS (B)        Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  ----------------  -----------  -------------  ----------------------------  -------------------------
ardusub          -3636 (-0.3985%)  0 (0.0000%)  4 (+0.0020%)   -3636 (-0.3981%)                                 122460
ardurover        -3836 (-0.4151%)  0 (0.0000%)  -4 (-0.0020%)  -3836 (-0.4147%)                                 111016
blimp            -2804 (-0.4011%)  0 (0.0000%)  -4 (-0.0020%)  -2804 (-0.4005%)                                 334888
arducopter-heli  -4192 (-0.4174%)  0 (0.0000%)  0 (0.0000%)    -4192 (-0.4170%)                                  31048
antennatracker   -2648 (-0.3580%)  0 (0.0000%)  0 (0.0000%)    -2648 (-0.3575%)                                 294172
arducopter       -3936 (-0.3964%)  0 (0.0000%)  0 (0.0000%)    -3936 (-0.3960%)                                  42176
arduplane        -4820 (-0.4704%)  0 (0.0000%)  4 (+0.0020%)   -4820 (-0.4699%)                                  11288
```

Hitec-Airspeed:
```
Binary Name    Text [B]         Data [B]     BSS (B)      Total Flash Change [B] (%)      Flash Free After PR (B)
-------------  ---------------  -----------  -----------  ----------------------------  -------------------------
ap_periph      -128 (-0.1351%)  0 (0.0000%)  0 (0.0000%)  -128 (-0.1344%)                                    9308
```

f103-GPS
```
Binary Name    Text [B]         Data [B]     BSS (B)      Total Flash Change [B] (%)      Flash Free After PR (B)
-------------  ---------------  -----------  -----------  ----------------------------  -------------------------
ap_periph      -232 (-0.2305%)  0 (0.0000%)  0 (0.0000%)  -232 (-0.2302%)                                    4912
```

File: ArduPlane/Parameters.cpp [hunk 1]
```diff
@@ -5,11 +5,11 @@
  *
  */
 
-#define GSCALAR(v, name, def) { plane.g.v.vtype, name, Parameters::k_param_ ## v, &plane.g.v, {def_value : def} }
-#define ASCALAR(v, name, def) { plane.aparm.v.vtype, name, Parameters::k_param_ ## v, (const void *)&plane.aparm.v, {def_value : def} }
-#define GGROUP(v, name, class) { AP_PARAM_GROUP, name, Parameters::k_param_ ## v, &plane.g.v, {group_info : class::var_info} }
-#define GOBJECT(v, name, class) { AP_PARAM_GROUP, name, Parameters::k_param_ ## v, (const void *)&plane.v, {group_info : class::var_info} }
-#define GOBJECTN(v, pname, name, class) { AP_PARAM_GROUP, name, Parameters::k_param_ ## pname, (const void *)&plane.v, {group_info : class::var_info} }
+#define GSCALAR(v, name, def)           { name, &plane.g.v,                   {def_value : def},              0, Parameters::k_param_ ## v,     plane.g.v.vtype }
```
Comment (tridge) [inline ArduPlane/Parameters.cpp:8]: better for these to move to AP_Param.h, and have a #define PARAM_VEHICLE_TYPE for the name of the vehicle object

**Review by tridge — CHANGES_REQUESTED** [review]

File: ArduPlane/Parameters.cpp (hunk 1 above)
Comment (IamPete1) [reply ArduPlane/Parameters.cpp]: Done

**Review by tridge — APPROVED** [review]

**Pull request merged**